### ISSUE TITLE
"paused" annotation not being respected for configurations

### ISSUE THREAD

**Issue by haarchri** [crossplane/crossplane#4619]
### What happened?
It appears that the "crossplane.io/paused: "true"" annotation is not being respected for configurations. 

### How can we reproduce it?
To reproduce this issue, follow these steps: create a configuration, add the "paused" annotation, and manually update a composition. After a reconciliation cycle of the configuration, the composition reverts back to its previous state.

### What environment did it happen in?
Crossplane version: v1.13.1

**Comment by negz** [comment]
Yeah, I don't think support was ever implemented for the package types.

I think there's another question here as to whether core types should support this annotation given that MRs are moving to  `spec.managementPolicy` instead. Should we support something like that on the core types too? CC @lsviben

**Comment by lsviben** [comment]
MRs are still supporting the pause annotation as well, at least for the time being. No plans for now to remove that and rely only on the `spec.managementPolicies` pausing.

When it comes to core types, I dont see a need for `managementPolicies` there, seems like an overkill. Just a simple pause annotation sounds good to me. Unless there are some use cases I am not aware of that could use more granularity?

**Comment by ytsarev** [comment]
After a discussion with @haarchri we identified two main supporting points to enable `pause` annotation for Configuration type

* Crossplane users are trying to migrate their Compositions to Upbound official providers and experimenting with the new manifests. Configuration reverting the migrated config with its own econciliation loop
* It will also be a great general enhancement enabling people to provide hotfixes and/or local experiments while Configuration is stopped. Otherwise, it is an uphill battle with the reconciliation loop.

We believe that adding 'pause' annotation support will enable easier Configuration adoption.

**Comment by negz** [comment]
Yeah, supporting this makes sense to me.

I actually thought no core types supported it, but a code search indicates that claim and XR does. So I think it's just the package manager controllers that need it.

I don't looove that there's two ways to pause an MR - this annotation and `managementPolicies`. I agree though that adding `managementPolicies` everywhere just to support pausing feels like overkill. I think my brain can handle a world where:

* Everything supports the `pause` annotation.
* MRs _also_ support a semantically equivalent `managementPolicies: []`.

**Comment by negz** [comment]
@haarchri @ytsarev Do either of you feel up to tackling this one? I _think_ it should be pretty straightforward.

**Comment by ytsarev** [comment]
> @haarchri @ytsarev Do either of you feel up to tackling this one? I think it should be pretty straightforward.

I would love to, but I don't know when I will have time for this one, definitely not this week. @haarchri let's sync up on the urgency of it.
